provider/src/lib.rs

```diff
@@ -30,12 +32,18 @@ thread_local! {
     static CONTEXT: RefCell<Context> = RefCell::new(Context::default())
 }
 
+#[cfg(target_family = "wasm")]
+thread_local! {
+    static OUTPUT_AND_LOG_PTRS: RefCell<[usize; 4]> = RefCell::new([0; 4]);
+}
+
 impl Default for Context {
     fn default() -> Self {
         Self {
             bump_allocator: Bump::new(),
             input_bytes: Vec::new(),
             output_bytes: ByteBuf::new(),
+            logs: Vec::with_capacity(1024),
```

Comment:
Given that we have now landed in explicit `initialize` / `finalize` function invocations, I wonder if we should make this configurable from the host?

Reply:
I think we briefly discuss that earlier on, maybe?

Reply:
I can pass it as an additional parameter in initialize. But also, it's a vector so it'll grow beyond the initial capacity if it needs to. But using the default capacity definitely hurts performance.

Reply:
Actually I guess another tradeoff is if the initial capacity changes, then the fuel consumption for a Function that performs logging may change as well. If we reduce the initial capacity, then a Function that performs enough logging to exceed the new capacity that would not have exceeded the old capacity would see an increase in fuel consumption. And if the initial capacity were increased, then a Function that performed enough logging to exceed the old capacity but not enough to exceed the new capacity would see a decrease in fuel consumption.